test/integration_test.py

```diff
@@ -644,6 +645,46 @@ def _manifest_filters(self, catalog: CatalogName) -> JSON:
             }
         }
 
+    def _test_manifest_tagging_race(self, catalog: CatalogName):
+        validators: dict[ManifestFormat, Callable[[CatalogName, bytes], None]] = {
+            ManifestFormat.compact: self._check_compact_manifest,
+            ManifestFormat.curl: self._check_curl_manifest,
+        }
+
+        for format, validator in validators.items():
+            with self.subTest('manifest_tagging_race', catalog=catalog, format=format):
+                filters = self._manifest_filters(catalog)
+                manifest_url = config.service_endpoint.set(path='/manifest/files',
+                                                           args=dict(catalog=catalog,
+                                                                     filters=json.dumps(filters),
+                                                                     format=format.value))
+                method = PUT
+                responses = []
+                while True:
+                    response = self._get_url(method, manifest_url)
+                    if response.status == 301:
+                        responses.append(response)
+                        # Request the same manifest without following the
+                        # redirect in order to expose the race condition that
+                        # returns an untagged cached manifest. This happens
+                        # when a step function execution has generated a
+                        # manifest but is still in the process of tagging the
+                        # object. The more requests we make, the more likely it
+                        # is that we catch the execution in this racy state. We
+                        # still have to throttle the requests in order to
+                        # prevent tripping the WAF rate limit.
+                        rate_limit = config.waf_rate_rule_period / config.waf_rate_rule_limit
```

Comment:
Time divided by count is not a rate. Rates are count divided by time. Just inline the variable.